### ISSUE TITLE
Article redirect tests

### ISSUE THREAD

**Issue by sareh** [bbc/simorgh#1704]
**Is your feature request related to a problem? Please describe.**

Where _:service_ is any service that should be served exclusively off .com (that is all non-UK services)

_NB scenarios below in strikethru relating to HTTPS uplift are out of scope till https://github.com/bbc/simorgh-infrastructure/issues/632 is resolved_

**Not .co.uk**

Given I go to https://www.bbc.co.uk/:service/articles/:id
Then I am given a 301 redirect to https://www.bbc.com/:service/articles/:id

~**Not http**~

~Given I go to http://www.bbc.co.uk/:service/articles/:id~
~Then I am given a 301 redirect to https://www.bbc.com/:service/articles/:id~

**AMP not on .co.uk**

Given I go to https://www.bbc.co.uk/:service/articles/:id.amp
Then I am given a 301 redirect to https://www.bbc.com/:service/articles/:id.amp

~**AMP not on http**~

~Given I go to http://www.bbc.co.uk/:service/articles/:id.amp~
~Then I am given a 301 redirect to https://www.bbc.com/:service/articles/:id.amp~

**Describe the solution you'd like**
- We should handle redirects within Mozart https://github.com/bbc/simorgh-infrastructure/issues/435. This ticket is only capturing the testing of redirects for our application. 
- These tests should only run against Test and Live environments, not local. 
- These can be run separately to the PR life cycle as a cron job.

Note: that until all of these services are routed on live, the tests should only be run against the test environment. An issue should be created to enable these on live when they are available.

**Describe alternatives you've considered**
We considered having the Cypress end-to-end tests in our current test suite repository, but decided against it - see the discussion thread below

**Testing notes**
We will need to add Cypress tests to make sure that the redirects happen as expected.
`Given I visit URL A, When I am redirected (served a 301 redirect), Then I will land on URL B`

Dev insight: Cypress tests will be sufficient.

**Comment by sareh** [comment]
Blocked on the infrastructure work: https://github.com/bbc/simorgh-infrastructure/issues/435

**Comment by dr3** [comment]
Is this not testing mozart behaviour? Which is what the mozart tests are for...

**Comment by DemonusPC** [comment]
Technically these are out e2e tests for the whole stack. So we should do a retest on our end to verify that the redirects are working fine for our product.

**Comment by jamesbhobbs** [comment]
We need to not do this. This would make our applications e2e tests and therefore our deploys dependent on code not in our application. This is a clear antipattern. I'd be ok with only expressing a warning rather than an error, but there is no means to do so in Cypress.

It is acceptable to have a different suite of Cypress tests for running tests such as this with 3rd party dependencies. We already have a similar example in a separate suite of tests for storybook, in .storybook/cypress. We could add some in something like 3rdPartyTests/cypress but we cannot add them in our primary suite. We would also need to ensure the tests in this additional suite are allowed to fail, i.e. only warn us of 3rd party errors.

**Comment by sareh** [comment]
I agree that we should not make our application PRs and merges dependent on the redirect status in Mozart. 

We do need some tests to warn us if the redirect behaviour changes for our pages, so we can prioritise fixes on the Mozart stack. Such tests only need to run against Test and Live environments, not local. Therefore, they can be run separately to the PR life cycle as a cron job.

**Comment by DemonusPC** [comment]
So do we close this issue? Or should the tests be created but separate to what we currently have?

**Comment by sareh** [comment]
@DemonusPC I've just updated the original description to reflect the outcome of the discussion in the thread. It might require more refinement by the infra pod prior to picking up. I'll leave that to your pod to decide.

**Comment by jamesbhobbs** [comment]
@DemonusPC It's now abs fine to do this in our 3rdPartyCypress tests, following the merge on the issue above. Still not v urgent though.

**Comment by jamesdonoh** [comment]
Following discussion with @jimjohnsonrollings and @benjaminhobbs have amended this issue to refer to all sites that require this behaviour, not just Persian.

**Comment by jamesdonoh** [comment]
Blocked on https://github.com/bbc/simorgh-infrastructure/issues/633

**Comment by jamesdonoh** [comment]
Unblocked as the above issue is done.
